flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/hybrid/tiered/tier/TierMemorySpec.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.io.network.partition.hybrid.tiered.tier;
+
+/**
+ * The memory specs for a tier, including the tier index, the number of exclusive buffers of the
+ * tier, whether the tier can use the shared buffers in the memory manager, etc.
+ */
+public class TierMemorySpec {
+
+    private final int tierIndex;
+
+    private final int numExclusiveBuffers;
+
+    private final boolean canUseSharedBuffers;
+
+    public TierMemorySpec(int tierIndex, int numExclusiveBuffers, boolean canUseSharedBuffers) {
+        this.tierIndex = tierIndex;
+        this.numExclusiveBuffers = numExclusiveBuffers;
+        this.canUseSharedBuffers = canUseSharedBuffers;
+    }
+
+    public int getTierIndex() {
+        return tierIndex;
+    }
+
+    public int getNumExclusiveBuffers() {
+        return numExclusiveBuffers;
+    }
+
+    public boolean canUseShareBuffers() {
+        return canUseSharedBuffers;
```

Comment:
Under what condition should this be `false`?

Reply:
Rename and refactor it to `TieredStorageMemorySpec`.
And `canUseSharedBuffers is renamed` to `isMemoryReleasable`.  

Some tiers(for example, the memory tier) should depend on the downstream consumption to release the memory. For these tiers, the `isMemoryReleasable` is `false`.